### PR TITLE
fix(channels): neutrino clients manual channels flagged private

### PR DESCRIPTION
**Pull request by JimmyMow**
This PR we make sure that if a user of ours is using neutrino that we flag their channels as private when they manually open them. Reason for this is because light consumer clients are not expected to have high uptime. 

However we cannot hardcode channels as private every time because we do support more power users driving their remote nodes or BTCPay Server with Zap. So we only flag channels as private if the `activeConnection` is `local`.

Addresses #681

### CODE REVIEW

**Comment by tanx** [comment]
👍

File: .flowconfig [hunk 1]
```diff
@@ -9,6 +9,7 @@
 <PROJECT_ROOT>/dll/.*
 <PROJECT_ROOT>/release/.*
 <PROJECT_ROOT>/git/.*
+<PROJECT_ROOT>/.git/.*
```
Comment (mrfelton) [inline .flowconfig]: Why is this needed?

File: app/lib/lnd/methods/channelController.js [hunk 2]
```diff
@@ -19,13 +19,14 @@ function ensurePeerConnected(lnd, pubkey, host) {
  * @return {[type]}         [description]
  */
 export function connectAndOpen(lnd, event, payload) {
-  const { pubkey, host, localamt } = payload
+  const { pubkey, host, localamt, private: privateChannel } = payload
```
Comment (mrfelton) [inline app/lib/lnd/methods/channelController.js]: Why do we need to change the variable name to `privateChannel`? Can't it just remain named `private`? especially as that is what we want it to be named when we pass it into `lnd.openChannel` below.

File: app/reducers/channels.js [hunk 3]
```diff
@@ -191,7 +192,18 @@ export const openChannel = ({ pubkey, host, local_amt }) => (dispatch, getState)
...
   dispatch(addLoadingPubkey(pubkey))
 
-  ipcRenderer.send('lnd', { msg: 'connectAndOpen', data: { pubkey, host, localamt } })
+  // Grab the activeConnection type from our local store.
+  // If the active connection type is local (light clients using neutrino)
+  // we will flag manually created channels as private. Other connections
+  // like remote node and BTCPay Server we will announce to the network as
+  // these users are using Zap to drive nodes that are online 24/7
+  const store = new Store({ name: 'settings' })
+  const { type } = store.get('activeConnection', null)
```
Comment (mrfelton) [inline app/reducers/channels.js]: If `activeConnection` is not set in the store, `store.get` should return an empty object, instead of `null`
 
```javascript
const { type } = store.get('activeConnection',{})
```

Otherwise, we would be trying to destructure `null` which would cause an error.

File: flow-typed/module_vx.x.x.js [hunk 4]
```diff
@@ -1,3 +1,3 @@
-declare module 'module' {
+declare module 'grpc' {
```
Comment (mrfelton) [inline flow-typed/module_vx.x.x.js]: We should probably create a new file for this rather than modifying the existing flowtype for `module`

File: test/unit/reducers/channels.spec.js [hunk 5]
```diff
@@ -8,6 +10,16 @@ import channelsReducer, {
   OPENING_FAILURE
 } from 'reducers/channels'
 
+jest.mock('electron', () => {
```
Comment (mrfelton) [inline test/unit/reducers/channels.spec.js]: This shouldn't be needed any longer as we've added a manual mock for electron here: https://github.com/LN-Zap/zap-desktop/blob/master/test/unit/__mocks__/electron.js

File: app/reducers/channels.js [hunk 6]
```diff
@@ -191,7 +192,18 @@ export const openChannel = ({ pubkey, host, local_amt }) => (dispatch, getState)
   dispatch(openingChannel())
   dispatch(addLoadingPubkey(pubkey))
 
-  ipcRenderer.send('lnd', { msg: 'connectAndOpen', data: { pubkey, host, localamt } })
+  // Grab the activeConnection type from our local store.
```
Comment (mrfelton) [inline app/reducers/channels.js]: nit: Line length should be 120.

**Comment by mrfelton** [comment]
@JimmyMow I've created a separate PR to address the issues with flow: https://github.com/LN-Zap/zap-desktop/pull/702. That can be removed from this PR.

File: flow-typed/module_vx.x.x.js (hunk 4 above)
Comment (JimmyMow) [reply flow-typed/module_vx.x.x.js]: going to remove this as this was fixed in a diff PR and merged

File: app/lib/lnd/methods/channelController.js (hunk 2 above)
Comment (JimmyMow) [reply app/lib/lnd/methods/channelController.js]: Because private is a reserved word in strict mode

File: test/unit/reducers/channels.spec.js (hunk 5 above)
Comment (JimmyMow) [reply test/unit/reducers/channels.spec.js]: 👍

**Comment by mrfelton** [comment]
Tested ACK https://github.com/LN-Zap/zap-desktop/pull/699/commits/9b837ba316133172389e8a338bfc6cb1055a9db1
